### PR TITLE
Abandoned crate loot refactor

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Refactored abandoned loot crates to use a weighted list instead of complex switch statement with hard-coded numbers.

Also replaced a duplicate entry for a bike horn with a spess knife.

## Why It's Good For The Game

The spawning logic was tricky and it was hard to adjust it. 

Now it is easier to add/change loot entries.

## Changelog

:cl:
refactor: Abandoned crates use weighted list and can yield spess knife now
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
this is honestly less readable than the shitcode switch version. funny

**Comment by MTandi** [comment]
> this is honestly less readable than the shitcode switch version. funny

It is all now in the `possible_loot` list, what do you mean?

**Comment by Helg2** [comment]
> It is all now in the `possible_loot` list, what do you mean?

Need to actually turn on your brain, quite hard as it is

File: code/modules/loot/loot.dm [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/**
+ * Datum for spawning multiple different atoms or applying some special spawning behavior
+ */
+/datum/loot
```
Comment (Jacquerel) [inline code/modules/loot/loot.dm:4]: I think calling this `/datum/loot` is too generic for what this is actually used for, it should specify that it is for the abandoned crate

Comment (Jacquerel) [reply code/modules/loot/loot.dm:4]: Or if you hope for this to be used more widely it should be more specific that it is for spawning items inside a container, which `loot` doesn't automatically imply

Comment (MTandi) [reply code/modules/loot/loot.dm:4]: It doesn't have to be a container, can work with a turf or anything else. So it is a generic datum.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Just scratch the loot datums and make them a subtype of `/obj/effect/spawner`, please. This is going to make things more confusing down the road.

**Pull request closed without merging**